src/Nest/Indices/MappingManagement/GetMapping/GetMappingResponse.cs

```diff
@@ -44,6 +44,13 @@ public class GetMappingResponse : DictionaryResponseBase<IndexName, IndexMapping
 		[JsonIgnore]
 		public IReadOnlyDictionary<IndexName, IndexMappings> Mappings => Indices;
 
+		[Obsolete("Use GetMappingFor explicitly instead this is a leaky abstraction that returns the mapping of the first index's first type on the response")]
+		public TypeMapping Mapping => this.Indices.FirstOrDefault().Value?.Mappings?.FirstOrDefault().Value;
+
+		public TypeMapping GetMappingFor<T>() => this.Indices[typeof(T)]?[typeof(T)];
```

Comment:
Is there any perf hit to doing a lookup of the same type? I'd assume this would be cached and also a micro optimizations.

Reply:
`typeof(T)` is not a compile time constant but yeah the IL for it is pretty minimal already :)